### PR TITLE
Implement Parse and TryParse static methods on Version and Range.

### PR DESCRIPTION
**Pull request by perfectsquircle**
This theoretically satisfies issue #10. Please review and let me know if I missed anything.

### CODE REVIEW

File: src/SemVer/Version.cs [hunk 1]
```diff
@@ -245,6 +245,45 @@ public override bool Equals(object other)
...
+        /// <param name="input">The version string.</param>
+        /// <returns>A boolean indicating success of the parse operation.</returns>
+        public static bool TryParse(string input, out Version result) 
+            => TryParse(input, loose: false, out result);
+
+        /// <summary>
+        /// Try to construct a new semantic version from a version string.
+        /// </summary>
+        /// <param name="input">The version string.</param>
+        /// <param name="loose">When true, be more forgiving of some invalid version specifications.</param>
```
Comment (adamreeve) [inline src/SemVer/Version.cs:273]: The two `Version.TryParse` methods are missing XML comment docs for the result parameter.

**Review by adamreeve — CHANGES_REQUESTED** [review]
Hi, looks good, thanks for this. Just a minor fix needed for the XML comments.

I'm on the fence about using overloads of `TryParse` with and without the loose parameter, it seems like having one method that takes an optional parameter would be better for consistency but I guess this was done to keep the output parameter as the last parameter? So happy to keep that as is.

File: src/SemVer/Version.cs (hunk 1 above)
Comment (perfectsquircle) [reply src/SemVer/Version.cs:273]: OK. I've pushed another commit with the missing XML docs.

**Comment by perfectsquircle** [comment]
> I guess this was done to keep the output parameter as the last parameter?

Yes. The compiler wants optional arguments at the end (CS1737), however it seems that convention for TryParse is that the `out` parameter appears at the end.

**Pull request merged**